### PR TITLE
:bug: Fix Metal3ClusterToMetal3Machines function in metal3Machine Controller

### LINKED ISSUE #40: The Metal3ClusterToMetal3Machines function returns a machine
**What steps did you take and what happened:**
The Metal3ClusterToMetal3Machines returns references to a machine, instead of a metal3machine. This is a remaining bit from v1alpha1.

**What did you expect to happen:**
It should return a reference to a Metal3Machine, hence m.Spec.InfrastructureRef.Name if set.

/kind bug

### PR DESCRIPTION
**Pull request by kashifest**
Metal3ClusterToMetal3Machines function was wrongly returning Machine object instead of metal3Machine. This PR fixes this bug and the relevant tests. 

Fixes #40

### CODE REVIEW

File: controllers/metal3machine_controller_test.go [hunk 1]
```diff
@@ -308,75 +307,90 @@ var _ = Describe("Metal3Machine manager", func() {
...
-		}
-		Expect(len(out)).To(Equal(2), "Expected 2 Metal3 machines to reconcile but got %d", len(out))
+	type TestCaseMetal3ClusterToBMM struct {
+		Cluster       *capi.Cluster
+		M3Cluster     *infrav1.Metal3Cluster
+		Machine0      *capi.Machine
+		Machine1      *capi.Machine
+		Machine2      *capi.Machine
+		M3Machine0    *infrav1.Metal3Machine
+		ExpectRequest int
```
Comment (Jaakko-Os) [inline controllers/metal3machine_controller_test.go]: I would understand 'ExpectedRequest' if it was 'bool'.
A comment would clarify the usage now when the type is 'int'. Below values 0,1,2 seem to be used. The meaning?

Comment (kashifest) [reply controllers/metal3machine_controller_test.go]: The number of requests returned by the function.

Comment (kashifest) [reply controllers/metal3machine_controller_test.go]: Changed it to bool.

File: controllers/metal3machine_controller_test.go [hunk 2]
```diff
@@ -308,75 +307,90 @@ var _ = Describe("Metal3Machine manager", func() {
...
-			machineNames[i] = out[i].Name
-		}
-		Expect(len(out)).To(Equal(2), "Expected 2 Metal3 machines to reconcile but got %d", len(out))
+	type TestCaseMetal3ClusterToBMM struct {
+		Cluster       *capi.Cluster
+		M3Cluster     *infrav1.Metal3Cluster
+		Machine0      *capi.Machine
+		Machine1      *capi.Machine
+		Machine2      *capi.Machine
+		M3Machine0    *infrav1.Metal3Machine
```
Comment (maelk) [inline controllers/metal3machine_controller_test.go]: you don't need this M3Machine0. The test is not fetching it

Comment (kashifest) [reply controllers/metal3machine_controller_test.go]: removed. thanks.

**Pull request merged**